pandas/core/internals/array_manager.py

```diff
@@ -999,11 +1000,16 @@ def _reindex_indexer(
 
         else:
             validate_indices(indexer, len(self._axes[0]))
+            indexer = ensure_int64(indexer)
```

Comment:
why are you converting to int64? we use intp for indexing via take

Reply:
That's existing code (just cut and paste). And our cython take algos always use int64:

https://github.com/pandas-dev/pandas/blob/d910a4435962505d3388d08a21e58bca6a9bf294/pandas/_libs/algos_take_helper.pxi.in#L69

Reply:
should they though?  i expect they ought to be intp_t

Reply:
im increasingly confident that intp_t is the correct thing to do, but its out of scope for this PR, xref #40390